modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -1454,16 +1456,32 @@ class SettingsImpl @Inject constructor(
     override fun getlastLoadedFromPodcastOrFilterUuid(): String? =
         getString(LAST_SELECTED_PODCAST_OR_FILTER_UUID)
 
-    override fun setBookmarksSortType(sortType: BookmarksSortType) {
-        setInt(Settings.PREFERENCE_BOOKMARKS_SORT, sortType.ordinal)
-        bookmarkSortTypeFlow.update { sortType }
+    override fun <T> setBookmarksSortType(sortType: T) {
+        when (sortType) {
+            is BookmarksSortTypeForPlayer -> {
+                setInt(Settings.PREFERENCE_BOOKMARKS_SORT_TYPE_FOR_PLAYER, sortType.ordinal)
+                bookmarkSortTypeForPlayerFlow.update { sortType }
+            }
+            is BookmarksSortTypeForPodcast -> {
+                setInt(Settings.PREFERENCE_BOOKMARKS_SORT_TYPE_FOR_PODCAST, sortType.ordinal)
+                bookmarkSortTypeForPodcastFlow.update { sortType }
+            }
+        }
     }
 
-    override fun getBookmarksSortType() =
-        BookmarksSortType.values()[
+    override fun getBookmarksSortTypeForPlayer() =
+        BookmarksSortTypeForPlayer.values()[
+            getInt(
+                Settings.PREFERENCE_BOOKMARKS_SORT_TYPE_FOR_PLAYER,
+                BookmarksSortTypeForPlayer.DATE_ADDED_NEWEST_TO_OLDEST.ordinal
+            )
+        ]
```

Comment:
I feel like persisting the enum's ordinal could make things difficult in the future, for example if we ever wanted to change the order of the items (since the order is currently tied to the order of the enums as well). It's also not super clear when debugging what a particular `Int` value means. What about storing it as a `String`, so it's not tied to the order? We could add something like a descriptive `persistenceKey: String` parameter to the enums and use that when we save it.

Reply:
Good point, I updated it in 97593af